web/src/selectors/index.js

```diff
@@ -5,9 +5,10 @@ import {
   sortBy,
   uniq,
 } from 'lodash';
+import { getCustomDatetime, getZoneId } from '../helpers/router';
 
 export function getSelectedZoneHistory(state) {
-  return state.data.histories[state.application.selectedZoneName] || [];
+  return state.data.histories[getZoneId()] || [];
```

Comment:
wouldn't it make sense to have a `getZoneId(state)` selector inside this file, instead of using router.js which imports store.js and uses getState()?
I think it's more logic to keep all the redux state selectors in this current file, but I might be wrong.

Reply:
You're right, there was something deeply wrong (and potentially buggy) with Redux selectors depending on the URL state helpers.

The reason I didn't want to support `getZoneId(state)` selector is because I wanted to get as close as possible to #2296 in this PR and that in particular means getting `zoneId` from the URL and not refer to its Redux state copy whenever possible.

I went to make all Redux selectors depend only on the Redux state in https://github.com/tmrowco/electricitymap-contrib/pull/2317/commits/69b73dd5e541c94e62d1dc91133afda844ef3e65, i.e I made them pure. I did that by wrapping them in _builder_ functions which accept all the dependencies passed to them straight from React components.

Please take a look and see what you think!